docs/site/Loopback-component-authorization.md

```diff
@@ -0,0 +1,549 @@
+[extension generator](./Extension-generator.md) guide for creating an extension.
+
+## Usage
+
+In order to use the above component into our REST API application, we have a few
+more steps to go.
+
+- Add component to application.
+
+{% include code-caption.html content="/src/application.ts" %}
+
+```ts
+this.component(AuthenticationComponent);
+```
+
+- Add permissions array to the role model.
+
+{% include code-caption.html content="/src/models/role.model.ts" %}
+
+```ts
+@model({
+  name: 'roles',
+})
+export class Role extends Entity {
+  // .....
+  // other attributes here
+  // .....
+
+  @property.array(String, {
+    required: true,
+  })
+  permissions: PermissionKey[];
+
+  constructor(data?: Partial<Role>) {
+    super(data);
+  }
+}
+```
+
+- Add user level permissions array to the user model. Do this if there is a use
+  case of explicit allow/deny of permissions at user-level in the application.
+  You can skip otherwise.
+
+{% include code-caption.html content="/src/models/user.model.ts" %}
+
+```ts
+@model({
+  name: 'users',
+})
+export class User extends Entity {
+  // .....
+  // other attributes here
+  // .....
+
+  @property.array(String)
+  permissions: UserPermission[];
+
+  constructor(data?: Partial<User>) {
+    super(data);
+  }
+}
+```
+
+- Add a step in custom sequence to check for authorization whenever any end
+  point is hit.
+
+{% include code-caption.html content="/src/sequence.ts" %}
+
+```ts
+import {inject} from '@loopback/context';
+import {
+  FindRoute,
+  InvokeMethod,
+  ParseParams,
+  Reject,
+  RequestContext,
+  RestBindings,
+  Send,
+  SequenceHandler,
+  HttpErrors,
+} from '@loopback/rest';
+import {AuthenticationBindings, AuthenticateFn} from './authenticate';
+import {
+  AuthorizatonBindings,
+  AuthorizeFn,
+  AuthorizeErrorKeys,
+} from './authorization';
+
+const SequenceActions = RestBindings.SequenceActions;
+
+export class MySequence implements SequenceHandler {
+  constructor(
+    @inject(SequenceActions.FIND_ROUTE) protected findRoute: FindRoute,
+    @inject(SequenceActions.PARSE_PARAMS) protected parseParams: ParseParams,
+    @inject(SequenceActions.INVOKE_METHOD) protected invoke: InvokeMethod,
+    @inject(SequenceActions.SEND) public send: Send,
+    @inject(SequenceActions.REJECT) public reject: Reject,
+    @inject(AuthenticationBindings.AUTH_ACTION)
+    protected authenticateRequest: AuthenticateFn,
+    @inject(AuthorizatonBindings.USER_PERMISSIONS)
+    protected fetchUserPermissons: UserPermissionsFn,
+    @inject(AuthorizatonBindings.AUTHORIZE_ACTION)
+    protected checkAuthorization: AuthorizeFn,
+  ) {}
+
+  async handle(context: RequestContext) {
+    try {
+      const {request, response} = context;
+      const route = this.findRoute(request);
+      const args = await this.parseParams(request, route);
+      // Do authentication of the user and fetch user permissions below
+      const authUser: AuthResponse = await this.authenticateRequest(request);
```

Comment:
@samarpanB Can you describe `AuthResponse` type? See #2896

Reply:
Done. Just didn't want to add authentication stuff to authorization doc. hence, skipped this.